SpatialGDK/Source/Private/Interop/GlobalStateManager.cpp

```diff
@@ -232,6 +234,16 @@ void UGlobalStateManager::GetSingletonActorAndChannel(FString ClassName, AActor*
 
 	OutActor = SingletonActorList[0];
 
+	// Check if actor is already processed
+	for (auto ClassChannelPair : NetDriver->SingletonActorChannels)
+	{
+		if (ClassChannelPair.Value.Key == OutActor)
+		{
+			OutChannel = ClassChannelPair.Value.Value;
```

Comment:
I would suggest to print a warning but since we're going to have inheritance soon it should be fine.